src/lib/pubkey/dilithium/dilithium_common/dilithium_polynomials.h

```diff
@@ -0,0 +1,1609 @@
+/*
+* Crystals Dilithium Digital Signature Algorithms
+* Based on the public domain reference implementation by the
+* designers (https://github.com/pq-crystals/dilithium)
+*
+* Further changes
+* (C) 2021-2022 Jack Lloyd
+* (C) 2021-2022 Manuel Glaser, Michael Boric, René Meusel - Rohde & Schwarz Cybersecurity
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#ifndef BOTAN_DILITHIUM_POLYNOMIAL_H_
+#define BOTAN_DILITHIUM_POLYNOMIAL_H_
+
+#include <botan/dilithium.h>
+
+#include <botan/internal/dilithium_symmetric_primitives.h>
+#include <botan/internal/shake.h>
+
+#include <array>
+#include <vector>
+#include <span>
+
+namespace Botan::Dilithium {
+
+class Polynomial
+   {
+   public:
+      std::array<int32_t, Botan::DilithiumModeConstants::N> m_coeffs;
```

Comment:
Does this really need to be public? Appears to be one user outside of `this`

Reply:
See: https://github.com/randombit/botan/pull/2973#discussion_r1073342589

Reply:
OK can you add a "// Public member variable" comment so that future readers know this is intentional *and* so we can easily find them in the future for cleanup purposes.